DynamicTablesPkg/Library/Acpi/RiscV/AcpiRhctLibRiscV/RhctGenerator.c

```diff
@@ -0,0 +1,997 @@
+         DEBUG_ERROR,
+         "ERROR: RHCT: Invalid Size of CMO Nodes. Status = %r\n",
+         Status
+        )
+        );
+      goto error_handler;
+    }
+
+    TableSize += NodeSize;
+
+    DEBUG (
+      (
+       DEBUG_INFO,
+       " CmoNodeCount = %d\n" \
+       " CmoOffset = %d\n",
+       CmoNodeCount,
+       CmoOffset
+      )
+      );
+  }
+
+  // Hart Info Nodes
+  if (HartInfoNodeCount > 0) {
+    HartInfoOffset = (UINT32)TableSize;
+    // Size of Hart Info node list.
+    NodeSize = GetSizeofHartInfoNodes (
+                 HartInfoOffset,
+                 RintcInfoNodeList,
+                 HartInfoNodeCount,
+                 RhctNodeCount - HartInfoNodeCount,
+                 &NodeIndexer
+                 );
+    if (NodeSize > MAX_UINT32) {
+      Status = EFI_INVALID_PARAMETER;
+      DEBUG (
+        (
+         DEBUG_ERROR,
+         "ERROR: RHCT: Invalid Size of Hart Info Nodes. Status = %r\n",
+         Status
+        )
+        );
+      goto error_handler;
+    }
+
+    TableSize += NodeSize;
+
+    DEBUG (
+      (
+       DEBUG_INFO,
+       " HartInfoNodeCount = %d\n" \
+       " HartInfoOffset = %d\n",
+       HartInfoNodeCount,
+       HartInfoOffset
+      )
+      );
+  }
+
+  DEBUG (
+    (
+     DEBUG_INFO,
+     "INFO: RHCT:\n" \
+     " RhctNodeCount = %d\n" \
+     " TableSize = 0x%x\n",
+     RhctNodeCount,
+     TableSize
+    )
+    );
+
+  if (TableSize > MAX_UINT32) {
+    Status = EFI_INVALID_PARAMETER;
+    DEBUG (
+      (
+       DEBUG_ERROR,
+       "ERROR: RHCT: RHCT Table Size 0x%lx > MAX_UINT32," \
+       " Status = %r\n",
+       TableSize,
+       Status
+      )
+      );
+    goto error_handler;
+  }
+
+  // Allocate the Buffer for RHCT table
+  *Table = (EFI_ACPI_DESCRIPTION_HEADER *)AllocateZeroPool (TableSize);
+  if (*Table == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    DEBUG (
+      (
+       DEBUG_ERROR,
+       "ERROR: RHCT: Failed to allocate memory for RHCT Table, Size = %d," \
+       " Status = %r\n",
+       TableSize,
+       Status
+      )
+      );
+    goto error_handler;
+  }
+
+  Rhct = (EFI_ACPI_6_6_RISCV_HART_CAPABILITIES_TABLE *)*Table;
+
+  DEBUG (
+    (
+     DEBUG_INFO,
+     "RHCT: Rhct = 0x%p TableSize = 0x%lx\n",
+     Rhct,
+     TableSize
+    )
+    );
+
+  Status = AddAcpiHeader (
+             CfgMgrProtocol,
+             This,
+             &Rhct->Header,
+             AcpiTableInfo,
+             (UINT32)TableSize
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG (
+      (
+       DEBUG_ERROR,
+       "ERROR: RHCT: Failed to add ACPI header. Status = %r\n",
+       Status
+      )
+      );
+    goto error_handler;
+  }
+
+  // Update RHCT table
+  Rhct->NodeCount    = RhctNodeCount;
+  Rhct->NodeOffset   = sizeof (EFI_ACPI_6_6_RISCV_HART_CAPABILITIES_TABLE);
+  Rhct->TimeBaseFreq = TimerInfo->TimeBaseFrequency;
+  Rhct->Flags        = TimerInfo->TimerCannotWakeCpu ? EFI_ACPI_6_6_RHCT_FLAG_TIMER_CANNOT_WAKEUP_CPU : 0;

```

Comment:
NIT:
Really up to you, but maybe it would be worth renaming:
`TimerInfo->TimerCannotWakeCpu` -> `TimerInfo->Flags`
in case more timer flags are added in the future.

In the other case, I think TimerCannotWakeCpu should be `UINT8` -> `BOOLEAN`